xray/core/plotting.py

```diff
@@ -0,0 +1,295 @@
+"""
+Plotting functions are implemented here and also monkeypatched into
+the DataArray class
+"""
+
+import functools
+
+import numpy as np
+
+from .utils import is_uniform_spaced
+
+
+# TODO - implement this
+class FacetGrid():
+    pass
+
+
+# Maybe more appropriate to keep this in .utils
+def _right_dtype(x, types):
+    """
+    Is x a sub dtype of anything in types?
+    """
+    return any(np.issubdtype(x.dtype, t) for t in types)
+
+
+def _ensure_plottable(*args):
+    """
+    Raise exception if there is anything in args that can't be plotted on
+    an axis.
+    """
+    plottypes = [np.floating, np.integer, np.timedelta64, np.datetime64]
+
+    # Lists need to be converted to np.arrays here.
+    if not any(_right_dtype(np.array(x), plottypes) for x in args):
+        raise TypeError('Plotting requires coordinates to be numeric '
+                        'or dates. Try DataArray.reindex() to convert.')
```

Comment:
`reindex` is not a method that converts string labels to numbers:
http://xray.readthedocs.org/en/stable/indexing.html#align-and-reindex

I would suggest `DataArray.assign_coords` to assign new coordinate labels.